src/components/Header/index.tsx

```diff
@@ -2,17 +2,17 @@ import React, { useState } from "react";
 import styles from "./index.module.scss";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faSearch } from "@fortawesome/free-solid-svg-icons";
-import { SearchForm } from "../SearchForm";
-import { OnSearch } from "../../containers/Main";
 
 type Props = {
-  onSearch: OnSearch;
+  searchForm: React.FunctionComponentElement<any>;
 };
 
-export const Header: React.FC<Props> = ({ onSearch }) => {
+export const SearchFormToggleContext = React.createContext(() => {});
```

Comment:
ref. https://ja.reactjs.org/docs/hooks-faq.html#how-to-avoid-passing-callbacks-down